ddl/ddl_api.go

```diff
@@ -4720,6 +4745,46 @@ func (d *ddl) AlterTableSetTiFlashReplica(ctx sessionctx.Context, ident ast.Iden
 		return errors.Trace(err)
 	}
 
+	// We should check this first, in order to avoid creating redundant DDL jobs.
+	tblInfo := tb.Meta()
+	if d.IsTiFlashPollEnabled() {
```

Comment:
Ditto question, why `// Must set placement rule before the ActionSetTiFlashReplica job is in queue.`?